anybox/recipe/odoo/vcs/git.py

```diff
@@ -406,6 +413,27 @@ def update_fetched_branch(self, branch):
                     self.log_call(['git', 'reset', '--hard', 'FETCH_HEAD'],
                                   callwith=update_check_call)
 
+    def _no_edit(self, cmd):
+        if self.git_version >= (1, 7, 10):
+            # --edit and --no-edit appear with Git 1.7.10
+            # see Documentation/RelNotes/1.7.10.txt of Git
+            # (https://git.kernel.org/cgit/git/git.git/tree)
+            cmd.insert(2, '--no-edit')
+        return cmd
+
+    def offline_merge(self, revision):
+        """Merge revision into current branch"""
+        if ishex(revision):
+            if not self.has_commit(revision):
+                raise UserError("Commit %s not found in git repository "
+                                "%s (offline mode)" % (revision, self))
+        elif not self._is_a_branch(revision):
+            raise UserError("Branch %s not found in git repository "
+                            "%s (offline mode)" % (revision, self))
+        with working_directory_keeper:
+            cmd = self._no_edit(['git', 'merge', revision])
```

Comment:
Instead of taking the `chdir` line from my suggested change you put the `_no_edit` call under the scope of `working_directory_keeper`. Have another look at my original comment. https://github.com/anybox/anybox.recipe.odoo/pull/132#pullrequestreview-585351239